spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java

```diff
@@ -328,4 +318,21 @@ protected void rollbackFromFileToListEnd(List<F> filteredFiles, F file) {
 
 	protected abstract boolean isDirectory(F file);
 
+	/**
+	 * Clear internal queue of listed files..
+	 */
+	public void clearUnprocessedFilesQueue() {
```

Comment:
OK. I cannot imaging the scenario when it is necessary, but let's give it a more meaningful name like `clearFetchedCache()` and some good Javadoc, plus `@since 6.3` and your name to the `@author` list of the affected classes.

I think another fix (separate issue/PR) must be done in the `StandardRotationPolicy` to enforce `setMaxFetchSize(1)` when mode is `fair`. Otherwise that fetched cache really does not make sense since we have just rotated to another dir and (possible) session factory. So, we would expect fresh file for that condition.